streaming/src/test/scala/org/apache/spark/streaming/util/WriteAheadLogSuite.scala

```diff
@@ -190,283 +281,150 @@ class WriteAheadLogSuite extends SparkFunSuite with BeforeAndAfter {
-
-    // Corrupt the first set of files so that they are basically unreadable
-    logFiles1.foreach { f =>
-      val raf = new FileOutputStream(f, true).getChannel()
-      raf.truncate(1)
-      raf.close()
-    }
-
-    // Verify that the corrupted files do not prevent reading of the second set of data
-    val readData = readDataUsingWriteAheadLog(testDir)
-    assert(readData === dataToWrite2)
-  }
-
-  test("FileBasedWriteAheadLog - do not create directories or files unless write") {
-    val nonexistentTempPath = File.createTempFile("test", "")
-    nonexistentTempPath.delete()
-    assert(!nonexistentTempPath.exists())
-
-    val writtenSegment = writeDataManually(generateRandomData(), testFile)
-    val wal = new FileBasedWriteAheadLog(new SparkConf(), tempDir.getAbsolutePath,
-      new Configuration(), 1, 1, closeFileAfterWrite = false)
-    assert(!nonexistentTempPath.exists(), "Directory created just by creating log object")
-    wal.read(writtenSegment.head)
-    assert(!nonexistentTempPath.exists(), "Directory created just by attempting to read segment")
+    assert(deaggregate.toSeq === events)
   }
-}
 
-object WriteAheadLogSuite {
+  test("BatchedWriteAheadLog - failures in wrappedLog get bubbled up") {
+    when(fileBasedWAL.write(any[ByteBuffer], anyLong)).thenThrow(new RuntimeException("Hello!"))
+    // the BatchedWriteAheadLog should bubble up any exceptions that may have happened during writes
+    val wal = new BatchedWriteAheadLog(fileBasedWAL, sparkConf)
 
-  class MockWriteAheadLog0() extends WriteAheadLog {
-    override def write(record: ByteBuffer, time: Long): WriteAheadLogRecordHandle = { null }
-    override def read(handle: WriteAheadLogRecordHandle): ByteBuffer = { null }
-    override def readAll(): util.Iterator[ByteBuffer] = { null }
-    override def clean(threshTime: Long, waitForCompletion: Boolean): Unit = { }
-    override def close(): Unit = { }
+    intercept[RuntimeException] {
+      val buffer = mock[ByteBuffer]
+      wal.write(buffer, 2L)
+    }
   }
 
-  class MockWriteAheadLog1(val conf: SparkConf) extends MockWriteAheadLog0()
-
-  class MockWriteAheadLog2(val conf: SparkConf, x: Int) extends MockWriteAheadLog0()
-
-
-  private val hadoopConf = new Configuration()
-
-  /** Write data to a file directly and return an array of the file segments written. */
-  def writeDataManually(data: Seq[String], file: String): Seq[FileBasedWriteAheadLogSegment] = {
-    val segments = new ArrayBuffer[FileBasedWriteAheadLogSegment]()
-    val writer = HdfsUtils.getOutputStream(file, hadoopConf)
-    data.foreach { item =>
-      val offset = writer.getPos
-      val bytes = Utils.serialize(item)
-      writer.writeInt(bytes.size)
-      writer.write(bytes)
-      segments += FileBasedWriteAheadLogSegment(file, offset, bytes.size)
-    }
-    writer.close()
-    segments
+  // we make the write requests in separate threads so that we don't block the test thread
+  private def eventFuture(
```

Comment:
I think this method can be available in a nice test utils trait. I can refactor those out later.